src/Liquid/Tag/TagExtends.php

```diff
@@ -100,12 +101,12 @@ private function findBlocks(array $tokens)
 	 *
 	 * @param array $tokens
 	 *
-	 * @throws \Liquid\LiquidException
+	 * @throws \Liquid\Exception\FilesystemException
 	 */
 	public function parse(array &$tokens)
 	{
 		if ($this->fileSystem === null) {
-			throw new LiquidException("No file system");
+			throw new FilesystemException("No file system");
```

Comment:
Don't you think it is a parsing error? "We are refusing to parse this because there's no filesystem!"

We could have added a null filesystem, whose responsibilities would be to throw a FilesystemException exception, and these exceptions would really *belong* to a kind of filesystem. But would it help to debug failing templates? Maybe not, maybe yes.